File: Solutions/0769/0769.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 0769.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-10-13 10:54:27
***************************************************************************************************/

int maxChunksToSorted(int* arr, int arrSize){
    int ret = 0;
    int mi = arrSize;
    int ma = 0;
    int l = 0;
    for (int idx=0; idx<arrSize; idx++) {
        int i = arr[idx];
        ma = i > ma ? i : ma;
        mi = i < mi ? i : mi;
        if (mi == l && ma == idx) {
            ret++;
            l = idx+1;
            mi = arrSize;
        }
    }
    return ret;
}

File: Solutions/0769/0769.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/10/13 10:48
# File: 0769.py
# Desc: 

class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        n = len(arr)
        # ret = 0
        # li = []
        # l = 0
        # for idx,i in enumerate(arr):
        #     li.append(i)
        #     if l == min(li) and idx == max(li):
        #         ret += 1
        #         li = []
        #         l = idx+1
        #     # print(li, l, idx)
        # return ret
        ret = 0
        mi = n
        ma = 0
        l = 0
        for idx, i in enumerate(arr):
            ma = max(i, ma)
            mi = min(i, mi)
            if mi == l and ma == idx: 
                ret += 1
                l = idx+1
                mi = n
            # else:
        return ret
